phpBB/styles/prosilver/template/posting_buttons.html

```diff
@@ -105,6 +105,16 @@
 	<button type="button" class="button button-icon-only bbcode-color" name="bbpalette" id="bbpalette" value="{L_FONT_COLOR}" onclick="change_palette();" title="{L_BBCODE_S_HELP}">
 		<i class="icon fa-tint fa-fw" aria-hidden="true"></i>
 	</button>
+	<!-- IF S_SMILIES_ALLOWED -->
+	<button type="button" class="button button-icon-only smiley_button" accesskey="y" name="addsmiley" title="{L_SMILEYS}">
```

Comment:
you do not seem to be using the correct structure for the dropdown which maybe why you need all this extra code

```
<div class="dropdown-container dropdown-button-control dropdown-right dropdown-down">
		<button type="button" class="button button-icon-only dropdown-trigger dropdown-select dropdown-toggle" accesskey="y" name="addsmiley" title="{L_SMILEYS}"
        		<i class="icon fa-smile-o fa-fw"></i>
		</button>
		<div class="dropdown" style="display: none; margin-right: -1293.38px;">
				<div class="pointer"><div class="pointer-inner"></div></div>
				<ul class="dropdown-contents dropdown-nonscroll smiley-list">
						<li>...
				</ul>
		</div>
</div>
```